fiberplane-templates/src/convert/mod.rs

```diff
@@ -198,26 +206,69 @@ fn print_cell(writer: &mut CodeWriter, cell: &Cell) {
                 ListType::Unordered => "listItem.unordered",
                 _ => panic!("Unknown ListType"),
             };
-            args.push(("content", format_content(&cell.content, &cell.formatting)));
+            args.push((
+                "content".to_string(),
+                format_content(&cell.content, &cell.formatting),
+            ));
             if let Some(level) = cell.level {
-                args.push(("level", level.to_string()));
+                args.push(("level".to_string(), level.to_string()));
             }
             if let Some(start_number) = cell.start_number {
-                args.push(("startNumber", start_number.to_string()));
+                args.push(("startNumber".to_string(), start_number.to_string()));
             }
             (function_name, cell.read_only)
         }
         Cell::Text(cell) => {
-            args.push(("content", format_content(&cell.content, &cell.formatting)));
+            args.push((
+                "content".to_string(),
+                format_content(&cell.content, &cell.formatting),
+            ));
             ("text", cell.read_only)
         }
         Cell::Provider(cell) => {
-            args.push(("title", escape_string(&cell.title)));
-            args.push(("intent", escape_string(&cell.intent)));
-            if let Some(query_data) = &cell.query_data {
-                args.push(("queryData", escape_string(query_data)));
-            }
-            ("provider", cell.read_only)
+            args.push(("title".to_string(), escape_string(&cell.title)));
+            let cell_type = match cell.intent.as_str() {
+                "prometheus,timeseries" => {
+                    if let Some(query) = decode_provider_cell_query_data(&cell.query_data) {
+                        args.extend(
+                            query
+                                .into_iter()
+                                .map(|(key, value)| (key, escape_string(value))),
+                        );
+                    }
+                    "prometheus"
+                }
+                "elasticsearch,events" => {
+                    if let Some(query) = decode_provider_cell_query_data(&cell.query_data) {
+                        args.extend(
+                            query
+                                .into_iter()
+                                .map(|(key, value)| (key, escape_string(value))),
+                        );
+                    }
+                    "elasticsearch"
+                }
+                "loki,events" => {
+                    if let Some(query) = decode_provider_cell_query_data(&cell.query_data) {
+                        args.extend(
+                            query
+                                .into_iter()
+                                .map(|(key, value)| (key, escape_string(value))),
+                        );
+                    }
+                    "loki"
+                }
```

Comment:
We should double-check the query data in the cells, otherwise we risk generating invalid templates. As a nice bonus, we'll be able to keep the first value in the `args` tuple a `&str`.

Something like:

```suggestion
                "prometheus,timeseries" | "elasticsearch,events" | "loki,events" => {
                    if let Some(query) = decode_provider_cell_query_data(&cell.query_data) {
                        for (key, value) in query.into_iter() {
                            match key => {
                                  "query" => args.push(("content", escape_string(value))),
                                  key => println!("Found unexpected key '{key}' in query data for cell with ID '{}'", cell.id()),
                            }
                        }
                        args.extend(
                            query
                                .into_iter()
                                .map(|(key, value)| (key, escape_string(value))),
                        );
                    }
                    cell.intent.split_once(',').unwrap().0
                }
```